src/mlpack/bindings/julia/julia_util.cpp

```diff
@@ -0,0 +1,510 @@
+  uint64_t* ints = new uint64_t[size];
+
+  for (size_t i = 0; i < size; ++i)
+    ints[i] = CLI::GetParam<std::vector<int>>(paramName)[i];
+
+  return ints;
+}
+
+/**
+ * Get the number of rows in a matrix parameter.
+ */
+size_t CLI_GetParamMatRows(const char* paramName)
+{
+  return CLI::GetParam<arma::mat>(paramName).n_rows;
+}
+
+/**
+ * Get the number of columns in a matrix parameter.
+ */
+size_t CLI_GetParamMatCols(const char* paramName)
+{
+  return CLI::GetParam<arma::mat>(paramName).n_cols;
+}
+
+/**
+ * Get the memory pointer for a matrix parameter.
+ * Note that this will assume that whatever is calling will take ownership of
+ * the memory!
+ */
+double* CLI_GetParamMat(const char* paramName)
+{
+  // Are we using preallocated memory?  If so we have to handle this more
+  // carefully.
+  arma::mat& mat = CLI::GetParam<arma::mat>(paramName);
+  if (mat.n_elem <= arma::arma_config::mat_prealloc)
+  {
+    // Copy the memory to something that we can give back to Julia.
+    double* newMem = new double[mat.n_elem];
+    arma::arrayops::copy(newMem, mat.mem, mat.n_elem);
+    return newMem; // We believe Julia will free it.  Hopefully we are right.
+  }
+  else
+  {
+    arma::access::rw(mat.mem_state) = 1;
+    return mat.memptr();
+  }
+}
+
+/**
+ * Get the number of rows in an unsigned matrix parameter.
+ */
+size_t CLI_GetParamUMatRows(const char* paramName)
+{
+  return CLI::GetParam<arma::Mat<size_t>>(paramName).n_rows;
+}
+
+/**
+ * Get the number of columns in an unsigned matrix parameter.
+ */
+size_t CLI_GetParamUMatCols(const char* paramName)
+{
+  return CLI::GetParam<arma::Mat<size_t>>(paramName).n_cols;
+}
+
+/**
+ * Get the memory pointer for an unsigned matrix parameter.
+ * Note that this will assume that whatever is calling will take ownership of
+ * the memory!
+ */
+size_t* CLI_GetParamUMat(const char* paramName)
+{
+  // Are we using preallocated memory?  If so we have to handle this more
+  // carefully.
+  arma::Mat<size_t>& mat = CLI::GetParam<arma::Mat<size_t>>(paramName);
+  if (mat.n_elem <= arma::arma_config::mat_prealloc)
+  {
+    // Copy the memory to something that we can give back to Julia.
+    size_t* newMem = new size_t[mat.n_elem];
+    arma::arrayops::copy(newMem, mat.mem, mat.n_elem);
+    return newMem; // We believe Julia will free it.  Hopefully we are right.
+  }
+  else
+  {
+    arma::access::rw(mat.mem_state) = 1;
+    return mat.memptr();
+  }
+}
+
+/**
+ * Get the number of rows in a column vector parameter.
+ */
+size_t CLI_GetParamColRows(const char* paramName)
+{
+  return CLI::GetParam<arma::vec>(paramName).n_rows;
+}
+
+/**
+ * Get the memory pointer for a column vector parameter.
+ * Note that this will assume that whatever is calling will take ownership of
+ * the memory!
+ */
+double* CLI_GetParamCol(const char* paramName)
+{
+  arma::access::rw(CLI::GetParam<arma::vec>(paramName).mem_state) = 1;
+  return CLI::GetParam<arma::vec>(paramName).memptr();
+}
```

Comment:
This might not be a big issue but still... The same situation as in #1949 - `arma::vec` might have its memory inlined since it inherits from `arma::Mat`. The same applies to `GetParamUCol`, `GeParamRow`, `GetParamURow` and `GetParamMatWithInfoPtr`

Reply:
You're right.  I thought that I checked all the overloads but apparently not ... fixed in 08ae85244.